packages/wonder-blocks-tooltip/util/is-obscured.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+/**
+ * Determine if an element is obscured by other elements.
+ *
+ * This uses document.elementFromPoint to see if the given element is being
+ * overdrawn by another element. Note that this won't work if the given element
+ * has `pointer-events: none`.
+ */
+export default function isObscured(element: Element) {
+    // TODO(somewhatabstract): We should be smarter in this algorithm and
+    // actually look at the intersection of the elements doing the obscuring
+    // just as we already do with our scroll parent intersections. That way we
+    // can not only check that the entire element is obscured, but think about
+    // partial obscurement so we can move the tooltip bubble when it's anchor
+    // point is not visible.
+
+    // Before we assume we're visible let's check to see if something else
+    // is obscuring us.  Here we make the assumption that if our topleft and
+    // bottomright corners are covered by something, we're not visible.
+    // There are ways that this can still not work, such as different
+    // elements only covering those points and the remainder being visible,
+    // or if some covering element has none for pointer-events style, but
+    // those edge cases shouldn't bother the main usages for this method.
+
+    // NOTE: If the anchor element has `pointer-events: none`, we're always
+    // going to end up hiding, so, you know, probably don't do that.
+    const anchorRect = element.getBoundingClientRect();
+    const style =
+        ((element: any).currentStyle: ?CSSStyleDeclaration) ||
+        window.getComputedStyle(element);
+    const anchorLeft =
+        anchorRect.left +
+        parseFloat(style.marginLeft) +
+        parseFloat(style.paddingLeft) +
+        parseFloat(style.borderLeftWidth);
```

Comment:
Would this be affected by `box-sizing`?

Reply:
I am not certain at this point. We should definitely test more edge cases (no pun intended).